src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,330 @@
-        super.client.newBuilder().addInterceptor(::imageIntercept).build()
+        /**
+         * Get cover image url from [element]
+         */
+        private fun getImageFromElement(element: Element): String {
+            val match = imageUrlRegex.find(element.selectFirst("source")!!.attr("data-srcset"))
+                ?: throw Exception("Cover image url not found")
+            return "https:" + match.value
+        }
+
+        private val DATE_PARSER by lazy { SimpleDateFormat("yyyy-MM-dd", Locale.JAPAN) }
+
+        /**
+         * Parse date from string like "3月31日" to UNIX Epoch time.
+         */
+        private fun String.toDate(): Long {
+            return DATE_PARSER.tryParse(this)
+        }
+    }
+
+    override val client = super.client.newBuilder()
+        .addNetworkInterceptor(::imageDescrambler)
+        .build()
 
-    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/ranking/manga")
 
-    // Show only ongoing works
-    override fun popularMangaSelector(): String = "ul[class=\"lineup-list ongoing\"] > li > div > a"
+    override fun popularMangaNextPageSelector() = null
 
-    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
-        title = element.select("h5").text()
-        thumbnail_url = element.select("div > img").attr("data-src")
-        setUrlWithoutDomain(element.attr("href"))
+    override fun popularMangaSelector() = ".ranking-item"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            url = element.selectFirst("a")!!.attr("href")
+            title = element.selectFirst(".title-text")!!.text()
+            author = element.selectFirst(".author-link")!!.text()
+            thumbnail_url = getImageFromElement(element)
+        }
     }
 
-    override fun latestUpdatesSelector() =
-        "div[class=\"update latest\"] > div.card-board > " + "div[class~=card]:not([class~=ad]) > div > a"
+    override fun mangaDetailsRequest(manga: SManga) = GET(manga.url)
 
-    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
-        title = element.select("div.data h3").text()
-        thumbnail_url = element.select("div.thumb-container img").attr("data-src")
-        setUrlWithoutDomain(element.attr("href"))
+    override fun mangaDetailsParse(document: Document): SManga {
+        val infoElement = document.selectFirst(".series-h-info")!!
+        val updateDateElement = infoElement.selectFirst("series-h-tag-label")
+        return SManga.create().apply {
+            title = infoElement.select("h1 > span")[1]!!.text()
+            author = infoElement.selectFirst(".series-h-credit-user-item > .article-text")!!.text() // TODO: get 脚本/漫画
+            description = infoElement.selectFirst(".series-h-credit-info-text-text > div > p > span > span")!!.text()
```

Comment:
Consider something like:
```kt
description = infoElement.selectFirst("#series-explain-area p").text()
```

Use `.wholeText().trim()` if you need to keep line breaks as-is.